lib/matplotlib/axes/_axes.py

```diff
@@ -212,6 +216,22 @@ def set_xlabel(self, xlabel, fontdict=None, labelpad=None, **kwargs):
         """
         if labelpad is not None:
             self.xaxis.labelpad = labelpad
+        _protected_kw = ['x', 'horizontalalignment', 'ha']
+        if any([k in kwargs for k in _protected_kw]):
+            if 'loc' in kwargs:
```

Comment:
I propse to introduce the parameter explicitly in the signature via:
~~~
set_xlabel(self, xlabel, fontdict=None, labelpad=None, *, loc=None, **kwargs):
~~~
because:
- then kwargs is really only "Text properties" as written in the docstring.
- explicit parameters can be used by code completers.
- `loc` is more high-level and thus should benefit from the visibility.

Reply:
Did i misunderstand what you were suggesting before?